### PR TITLE
AzureMonitor: Check to see if response is nil to prevent panic

### PR DESCRIPTION
**Pull request by bossinc**
This will prevent panics from happening when there is no response. 
https://github.com/grafana/support-escalations/issues/6911

### CODE REVIEW

**Review by aangelisc — COMMENTED** [review]
~~Is the problem that `res` is `nil` or that `res.Body` is? Looking at the logs the issue is on line 232 (which is 274 here) so maybe we should be checking both?~~ NVM me I double checked this is good to go 😊

**Review by aangelisc — APPROVED** [review]

**Review by aangelisc — CHANGES_REQUESTED** [review]
Ok I'm retracting my initial review because I'm an idiot. If `res` is `nil` that means that `err` is not `nil`. So we can move the return into the error handling condition above rather than having a separate condition. We can also fix the log statement on 267 as it seems the interpolation isn't working so the true underlying API error is hidden in the logs 😊

File: pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go [hunk 1]
```diff
@@ -346,7 +342,11 @@ func (e *AzureMonitorDatasource) executeQuery(ctx context.Context, logger log.Lo
 		return dataResponse
 	}
 
-	subscription := e.retrieveSubscriptionDetails(cli, ctx, logger, tracer, query.Subscription, dsInfo.Routes["Azure Monitor"].URL, dsInfo.DatasourceID, dsInfo.OrgID)
+	subscription, err := e.retrieveSubscriptionDetails(cli, ctx, logger, tracer, query.Subscription, dsInfo.Routes["Azure Monitor"].URL, dsInfo.DatasourceID, dsInfo.OrgID)
+	if err != nil {
+		dataResponse.Error = err
```
Comment (aangelisc) [inline pkg/tsdb/azuremonitor/metrics/azuremonitor-datasource.go]: I think we should just log the error here at this point and continue with the query, wdyt?

**Review by aangelisc — APPROVED** [review]

**Review by asimpson — APPROVED** [review]
Like this approach, nice work :trophy:

**Pull request merged**